src/logics/ArithLogic.cc

```diff
@@ -400,8 +429,10 @@ lbool ArithLogic::arithmeticElimination(vec<PTRef> const & top_level_arith, Subs
     std::transform(top_level_arith.begin(), top_level_arith.end(), std::back_inserter(polynomials),
                    [&](auto const & eq) { return ptrefToPoly(eq, logic); });
 
-    auto constSubstitutions = collectConstantSubstitutions(logic, polynomials);
-    for (PTRef key : constSubstitutions.getKeys()) {
+    auto substitutionsOrConflict = collectConstantSubstitutions(logic, polynomials);
+    if (substitutionsOrConflict.isConflict()) { return l_False; }
+    for (SubstMap const constSubstitutions = substitutionsOrConflict.extractSubstitutions();
+         PTRef key : constSubstitutions.getKeys()) {
```

Comment:
```suggestion
         PTRef key : substitutionsOrConflict.getSubstitutions().getKeys()) {
```
Another possibility. Probably no need to destroy it?

Reply:
For simplicity I removed the destructive method and only kept and used `getSubstitutions`.